cmd/gpu_plugin/gpu_plugin_test.go

```diff
@@ -696,3 +714,253 @@ func TestBypath(t *testing.T) {
 		}
 	}
 }
+
+func TestPciDeviceForCard(t *testing.T) {
+	root, err := os.MkdirTemp("", "test_pci_device_for_card")
+	if err != nil {
+		t.Fatalf("can't create temporary directory: %+v", err)
+	}
+	// dirs/files need to be removed for the next test
+	defer os.RemoveAll(root)
+
+	sysfs := path.Join(root, "sys")
+
+	cardPath := filepath.Join(sysfs, "class", "drm", "card0")
+	cardDevicePath := filepath.Join(cardPath, "device")
+
+	if err := os.MkdirAll(cardDevicePath, 0750); err != nil {
+		t.Fatalf("couldn't create card's device path")
+	}
+
+	data := "0x5959"
+
+	err = os.WriteFile(filepath.Join(cardDevicePath, "device"), []byte(data), 0o600)
+	if err != nil {
+		t.Fatalf("device id write failed: %+v", err)
+	}
+
+	id, err := pciDeviceIDForCard(cardPath)
+
+	if err != nil {
+		t.Error("failed to get device id for card")
+	}
+
+	if id != data {
+		t.Errorf("wrong id received %s vs %s", id, data)
+	}
+
+	// Check bad device
+
+	cardPath = filepath.Join(sysfs, "class", "drm", "card1")
+	cardDevicePath = filepath.Join(cardPath, "device")
+
+	if err := os.MkdirAll(cardDevicePath, 0750); err != nil {
+		t.Fatalf("couldn't create card's device path")
+	}
+
+	err = os.WriteFile(filepath.Join(cardDevicePath, "devicee"), []byte(data), 0o600)
+	if err != nil {
+		t.Fatalf("device id write failed: %+v", err)
+	}
+
+	id, err = pciDeviceIDForCard(cardPath)
+
+	if err == nil {
+		t.Errorf("id received when it shouldn't be possible: %s", id)
+	}
+}
+
+type symlinkItem struct {
+	old string
+	new string
+}
+
+func createSymlinks(t *testing.T, base string, links []symlinkItem) {
+	for _, link := range links {
+		linkOld := filepath.Join(base, link.old)
+		linkNew := filepath.Join(base, link.new)
+
+		if _, err := os.Stat(linkOld); err != nil {
+			if err := os.MkdirAll(linkOld, 0o750); err != nil && !errors.Is(err, os.ErrExist) {
+				t.Fatalf("failed to create symlink base dir: %+v", err)
+			}
+		}
+
+		d := filepath.Dir(linkNew)
+		if err := os.MkdirAll(d, 0o750); err != nil {
+			t.Fatal("failed to create symlink new dir", err)
+		}
+
+		if err := os.Symlink(linkOld, linkNew); err != nil {
+			t.Fatal("failed to create symlink from old to new", err)
+		}
+	}
+}
+
+func createFiles(t *testing.T, base string, files map[string][]byte) {
+	for file, content := range files {
+		fp := filepath.Join(base, file)
+		dir := filepath.Dir(fp)
+
+		if err := os.MkdirAll(dir, 0o750); err != nil {
```

Comment:
Some error messages start with "failed", some with "Failed", some have that in start, some in end... Consistency?

Reply:
Went through all the Error/Fatal calls and changed them to start with upper case.

Reply:
There's still the "Failed something" and "Something failed" inconsistency, but it's not that important.  Rest seems OK on quick glance.